libavpipe/src/avpipe_mux.c

```diff
@@ -228,6 +228,12 @@ avpipe_init_muxer(
     char *out_filename = p->url;
     xctx_t *p_xctx;
 
+    in_mux_ctx->video.parts = (char **) calloc(MAX_MUX_IN_STREAM, sizeof(char *));
```

Comment:
Do we know at all how many streams there may be at this point? It seems a little overkill to allocate 16MB of pointers to parts for every single transcode, when the vast majority of the time only a small handful of the streams are used.

Reply:
No, we need to preprocess to pass the number of streams and their index (using some params). This is the most straight forward solution.